### PR TITLE
identity: introspect basearch

### PR DESCRIPTION
**Pull request by lucab**
This introspects node basearch from rpm-ostree, instead of using
build-time configuration and the internal mapping table.

Ref: https://github.com/coreos/coreos-assembler/pull/590
Closes: https://github.com/coreos/zincati/issues/54

### CODE REVIEW

File: src/rpm_ostree/cli_status.rs [hunk 1]
```diff
@@ -18,6 +18,8 @@ pub struct DeploymentJSON {
     booted: bool,
     #[serde(rename = "base-checksum")]
     base_checksum: Option<String>,
+    #[serde(rename = "coreos-assembler.basearch")]
```
Comment (jlebon) [inline src/rpm_ostree/cli_status.rs]: Check out the JSON at https://github.com/coreos/zincati/issues/54#issuecomment-508087569. This is actually nested under `base-commit-meta`.

**Comment by lucab** [comment]
Ouch, I totally missed one level of nesting. To compensate for that, I've know added fixtures and mock to cover the deserialization part, PTAL

File: tests/fixtures/rpm-ostree-status.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        "rpmostree.inputhash" : "0e2ba98bac847c8a8cc5ffd42822a41d53ef6719ccaf6a0094559768aa24da33",
+        "version" : "30.1",
+        "rpmostree.rpmmd-repos" : [
+          {
+            "id" : "fedora-coreos-pool",
+            "timestamp" : 7435743550895030272
+          },
+          {
+            "id" : "fedora",
+            "timestamp" : -7689964138319052800
```
Comment (lucab) [inline tests/fixtures/rpm-ostree-status.json]: @jlebon not strictly related to this PR, but I fear there something going wrong in the compose if this timestamp ends up negative.

Comment (jlebon) [reply tests/fixtures/rpm-ostree-status.json]: Yeah, that's tracked in https://github.com/projectatomic/rpm-ostree/issues/1734. Need to look at that again soon.

**Pull request merged**